packages/node-cli/src/CLI.js

```diff
@@ -49,18 +54,49 @@ module.exports = class CLI {
 
 			program
 				.version(CLI_VERSION, '--version')
-				.option(`${INTERACTIVE_ALIAS}, ${INTERACTIVE_OPTION}`, NodeTranslationService.getMessage(INTERACTIVE_OPTION_DESCRIPTION))
-				.on('command:*', args => {
+				.option(
+					`${INTERACTIVE_ALIAS}, ${INTERACTIVE_OPTION}`,
+					NodeTranslationService.getMessage(INTERACTIVE_OPTION_DESCRIPTION),
+					this._validateInteractive
+				)
+				.on('command:*', (args) => {
 					NodeConsoleLogger.error(NodeTranslationService.getMessage(ERRORS.COMMAND_DOES_NOT_EXIST, args[0]));
 				})
 				.usage(NodeTranslationService.getMessage(USAGE))
 				.parse(process.argv);
-
 		} catch (exception) {
 			NodeConsoleLogger.error(unwrapExceptionMessage(exception));
 		}
 	}
 
+	_validateCommandExists(commandMetadataList, commandName) {
+		if (commandName && !commandMetadataList.hasOwnProperty(commandName) && commandName !== HELP_ALIAS && commandName !== HELP_OPTION) {
+			throw NodeTranslationService.getMessage(ERRORS.COMMAND_DOES_NOT_EXIST, commandName);
+		}
+	}
+
+	_takeOutUnnecesaryArgumentsWhenHelpIsCalled() {
+		if (process.argv.includes(HELP_ALIAS) || process.argv.includes(HELP_OPTION)) {
+			if (process.argv.length > 3) {
+				// In case there is a command
+				return process.argv.slice(0, 3).concat(HELP_ALIAS);
+			}
+			return process.argv.slice(0, 2);
```

Comment:
arg[0] = node.exe
arg[1] = suitecloud.js
arg[2] = existingCommandName
+ HELP_ALIAS or HELP_OPTINON
These ones already sum 4.
We are always going to get into the if statement.

Reply:
But this is fine. Let me explain with an example:

if we have commandName --arg1 --arg2 --help, we want all arguments deleted except the --help one... 
if we have commandName --help --arg2 --arg3, the same
if we have commandName --help, we would still want help appearing.

So, this way, we ensure that, if there is a help argument and more than 3 arguments (i.e we write a command), we would show the help of the particular command.

If there is only two arguments, it's because it is the general help, without command, i.e suitecloud --help.

Reply:
I'm changing it a little bit though so it would be simplified.